src/posix/platform/config_file.cpp

```diff
@@ -44,10 +44,15 @@ namespace ot {
 namespace Posix {
 
 ConfigFile::ConfigFile(const char *aFilePath)
-    : mFilePath(aFilePath)
 {
-    assert(mFilePath != nullptr);
-    VerifyOrDie(strlen(mFilePath) + strlen(kSwapSuffix) < kFilePathMaxSize, OT_EXIT_FAILURE);
+    SetFilePath(aFilePath);
+}
+
+void ConfigFile::SetFilePath(const char *aFilePath)
+{
+    assert(aFilePath != nullptr);
+    VerifyOrDie(strlen(aFilePath) + strlen(kSwapSuffix) < kFilePathMaxSize, OT_EXIT_FAILURE);
+    mFilePath = aFilePath;
```

Comment:
The original code uses the `mFilePath` to point to the file path, because the file path is a pre-defined macro. The space of the file path string will never be released. In this commit, the file path could be dynamic specified, we can't guarantee that the memory pointed by the `aFilePath` won't be released. I think we may need to store the file path like the [code](https://github.com/openthread/openthread/blob/main/src/posix/platform/settings_file.cpp#L58), rather than using a pointer to point to the file path.